### DISCUSSION TITLE
Default Sanity dev command broken

### DISCUSSION THREAD

**Discussion by unitof** [vercel/vercel#8511] [Ideas] [answered]
I'm finding that the default `dev` command for Sanity projects (`sanity start --port $PORT`) is broken when used under `vercel dev` (at least on on macOS 12.5, on an M1 machine).

The reason appears to be that Vercel starts Sanity on its chosen port on **IPv6**, but only proxies localhost:3000 to IPv4.

I have a [PR drafted](https://github.com/vercel/vercel/pull/8512) which fixes this issue for Sanity projects by adding `--host 0.0.0.0` to the default Sanity dev command (forcing IPv4), but could use some guidance whether this is a deeper `vercel dev` bug, or perhaps an upstream Sanity one. Submitting here first, as requested.

Issue & solution found thanks to https://github.com/vercel/vercel/discussions/6639; unclear if that was with Sanity or another tool (but appears to be a different tool).

## `vercel dev` running with current default Sanity dev command (`sanity start --port $PORT`)
```
$ lsof -nP -iTCP -sTCP:LISTEN
⋮
node      73286 jacob   26u  IPv6 0xb13c8e406decedcd      0t0  TCP *:3000 (LISTEN)
node      73324 jacob   21u  IPv6 0xb13c8e406e5e42cd      0t0  TCP [::1]:58335 (LISTEN)
```
Result: loading http://localhost:3000 never proxies to Sanity, eventually times out in browser

## `vercel dev` with dev command changed to `sanity start --host 0.0.0.0 --port $PORT`

```
$ lsof -nP -iTCP -sTCP:LISTEN
⋮
node      73199 jacob   26u  IPv6 0xb13c8e406bb317cd      0t0  TCP *:3000 (LISTEN)
node      73233 jacob   22u  IPv4 0xb13c8e4a0a161c45      0t0  TCP 127.0.0.1:58178 (LISTEN)
```
Result: http://localhost:3000 loads Sanity as expected, proxying to port 58178.

**Comment by unitof** [comment]
Hi all! Any feedback on this?

**Comment by amyegan** [comment]
Hi @unitof! Thanks for putting this together! It looks like a great solution to me, but I don't have the background knowledge to say what else could be affected by your proposed change. I asked the CLI engineers to take a look at it for us when they can spare some time.

**Comment by unitof** [comment]
Thanks @amyegan! No rush.

The [PR](#8512) I opened does indeed fix this on Sanity projects, but it looks like it may actually be an upstream bug with `vc dev` itself, at least on macOS, starting processes on an IPv6 proxied port instead of the expected IPv4.

It apparently happens on plain React projects as well: https://github.com/vercel/vercel/discussions/6639#discussioncomment-3610190

**Comment by styfle** [comment, marked as answer]
Thanks! This sounds similar to a recent issue with `vite`.

We'll want to solve this in `vc dev` for all frameworks. See my comment here https://github.com/vercel/vercel/issues/8121#issuecomment-1226454969

**Reply by unitof** [reply]
Sounds good to me! Thanks. Will eagerly await `vc dev` supporting proxying to IPv6 (or help if I can!).

**Reply by amyegan** [reply]
New PR to address this issue 😁 https://github.com/vercel/vercel/pull/8542
